apis/v1alpha1/shared_types.go

```diff
@@ -80,77 +96,111 @@ type Port struct {
+// Exactly one of the selector pointers must be set for a given peer. If a
+// consumer observes none of its fields are set, they must assume an unknown
+// option has been specified and fail closed.
 // +kubebuilder:validation:MaxProperties=1
 // +kubebuilder:validation:MinProperties=1
-type NamespacedPeer struct {
-	// NamespaceSelector is a labelSelector used to select Namespaces, This field
-	// follows standard label selector semantics; if present but empty, it selects
-	// all Namespaces.
+type AdminNetworkPolicyEgressPeer struct {
+	// Namespaces defines a way to select all pods within a set of Namespaces.
+	// Note that host-networked pods are not included in this type of peer.
+	//
+	// Support: Core
+	//
 	// +optional
-	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
-
-	// SameLabels is used to select a set of Namespaces that share the same values
-	// for a set of labels.
-	// To be selected a Namespace must have all of the labels defined in SameLabels,
-	// AND they must all have the same value as the subject of this policy.
-	// If Samelabels is Empty then nothing is selected.
+	Namespaces *metav1.LabelSelector `json:"namespaces,omitempty"`
+	// Pods defines a way to select a set of pods in
+	// a set of namespaces. Note that host-networked pods
+	// are not included in this type of peer.
+	//
+	// Support: Core
+	//
+	// +optional
+	Pods *NamespacedPod `json:"pods,omitempty"`
+	// Nodes defines a way to select a set of nodes in
+	// the cluster. This field follows standard label selector
+	// semantics; if present but empty, it selects all Nodes.
+	//
+	// Support: Extended
+	//
+	// <network-policy-api:experimental>
 	// +optional
-	// +kubebuilder:validation:MaxItems=100
-	SameLabels []string `json:"sameLabels,omitempty"`
-
-	// NotSameLabels is used to select a set of Namespaces that do not have certain
-	// values for a set of label(s).
-	// To be selected a Namespace must have all of the labels defined in NotSameLabels,
-	// AND at least one of them must have different values than the subject of this policy.
-	// If NotSameLabels is empty then nothing is selected.
+	Nodes *metav1.LabelSelector `json:"nodes,omitempty"`
+	// Networks defines a way to select peers via CIDR blocks.
+	// This is intended for representing entities that live outside the cluster,
+	// which can't be selected by pods, namespaces and nodes peers, but note
+	// that cluster-internal traffic will be checked against the rule as
+	// well. So if you Allow or Deny traffic to `"0.0.0.0/0"`, that will allow
+	// or deny all IPv4 pod-to-pod traffic as well. If you don't want that,
+	// add a rule that Passes all pod traffic before the Networks rule.
+	//
+	// Each item in Networks should be provided in the CIDR format and should be
+	// IPv4 or IPv6, for example "10.0.0.0/8" or "fd00::/8".
+	//
+	// Networks can have upto 25 CIDRs specified.
+	//
+	// Support: Extended
```

Comment:
More "Extended". I think this was supposed to be "Experimental", because it's expected to eventually be Core but it's not widely implemented enough yet?